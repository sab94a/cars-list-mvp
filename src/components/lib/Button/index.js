// @flow

import React from 'react';
import type { Button as Props} from 'components/lib/types'
import styles from './index.module.scss';

const Button = ({ children, ...rest }: Props) => (
    <button {...rest } className={ styles.root } >{ children }</button>
);

export default React.memo<Props>(Button)
