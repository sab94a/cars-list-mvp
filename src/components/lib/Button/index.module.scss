@import 'components/lib/variables.scss';

.root {
    border: 0;
    background: $primary;
    color: #fff;
    width: $button-width;
    font-size: $font-size-sm;
    padding: $offset-sm;
    border-radius: 2px;
    cursor: pointer;

    &:hover, &:focus {
        background: $primary-pressed;
    }
}