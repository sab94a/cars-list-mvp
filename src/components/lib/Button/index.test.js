import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './'

describe('<Button />', () => {
    it('Should render correctly', () => {
        const component = mount(<Button />)

        expect(toJson(component)).toMatchSnapshot();
    });
});
