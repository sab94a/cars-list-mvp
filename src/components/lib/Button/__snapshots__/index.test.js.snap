// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render correctly 1`] = `
<Memo(Button)>
  <button
    className="root"
  />
</Memo(Button)>
`;
